js/src/components/SidebarSnippetPreviewButton.js

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import { ButtonSection } from "yoast-components";
+import { Button, Modal } from "@wordpress/components";
+import { __ } from "@wordpress/i18n/build/index";
+import SnippetEditorWrapper from "../containers/SnippetEditor";
+
+class SnippetPreviewModal extends React.Component {
```

Comment:
I'm not sure about the naming of/within this file. The filename is `SidebarSnippetPreviewButton`, while the classname is `SnippetPreviewModal`. In reality, this file contains the render for both a button and a modal. Let's discuss naming together.